[BFN] Implementation API for platform component

taras-keryk opened the pull request:
Signed-off-by: Taras Keryk <tarasx.keryk@intel.com>

	These changes are needed to read the version of the BIOS and BMC component.

#### Why I did it
    The previous implementation didn't have platform components API, so fwutil return an empty list.
    After implementation of the platform component API, we have actual list of platform components and firmware 
#### How I did it
    Create components.py module, updated platform.json and platform_components.json
    Add function for reading BMC version to thrift interface

#### How to verify it
    Run manually 'fwutil show status' or run unit tests

    Previous 'fwutil show status' had no information output
    New  'fwutil show status' command output:
                    Chassis                   Module    Component    Version    Description
                    ------------------------  --------  -----------  ---------  -------------
                    Chassis1                  N/A       BIOS         1.2.3     Chassis BIOS
                                                        BMC          5.1       Chassis BMC
#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

Commit:
[BFN] Implementation API for platform component
	SONiC has a concept of "platform components"
	this may include - CPLD, FPGA, BIOS, BMC, etc.

	These changes are needed to read the version of the BIOS and BMC component.

	What I did
		Create components.py module
		Add funcion for reading componet version to thrift interface
	How I did it
		The previous implementaion didn't have platform components API, so fwutil return an empty list.
		After implementation of the platform component API, we have actual list of platform components and firmware versions

	How to verify it
		Run manually 'fwutil show status' or run unit tests

	Previous command output
		Chassis                   Module    Component    Version    Description
		------------------------  --------  -----------  ---------  -------------

	New command output
		Chassis                   Module    Component    Version    Description
                ------------------------  --------  -----------  ---------  -------------
                Chassis1		  N/A       BIOS         1.2.3	   Chassis BIOS
                                                    BMC          5.1	   Chassis BMC
Signed-off-by: Taras Keryk <tarasx.keryk@intel.com>

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/chassis.py:
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env python
 
+
```
please remove empty lines here and below

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/chassis.py:
```diff
@@ -289,6 +301,9 @@ def get_reboot_cause(self):
         """
         return self.REBOOT_CAUSE_NON_HARDWARE, ''
 
+    def get_all_components(self):
```
implemented in base class
https://github.com/Azure/sonic-platform-common/blob/5da31e190d0df1ba58078cf182f6a3b3918253ab/sonic_platform_base/chassis_base.py#L169

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/chassis.py:
```diff
@@ -333,3 +348,15 @@ def get_status_led(self):
         """
         return self.system_led
+
+    def get_num_components(self):
```
implemented in base class

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+class Components(ComponentBase):
+    """BFN Montara Platform-specific Component class"""
+
+    #bf_model = get_model()
```
please remove all debug information

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the FAN
```
please fix the description

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+        return False
+    
+    def get_available_firmware_version(self, image_path):
+            return 'N/A'
```
please fix indentation

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+            A string containing the component firmware update notification if required.
+            By default 'None' value will be used, which indicates that no actions are required
+        """
+        return 'N/A'
```
shouldn't it be `None` as mentioned in the method description?

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+        Raises:
+            RuntimeError: update failed
+        """
+        return None
```
shouldn't it raise `RuntimeError` as mentioned in the description?

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+    import os
+    stream = os.popen('show version  | grep Platform | awk \'{print $2}\'')
+    return stream.read()
+    #return subprocess.check_output(['show', 'version', '|', 'grep', 'Platform', '|', 'awk', '\'{print $2}\'']).strip()
```
please remove all debug information

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env python
+
+########################################################################
```
Please remove this header if not used in other plugins

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+        self.__component_list = []
+        self.__bf_model = bf_model
+    """
+    def __get_platform_components_path(self, root_path):
```
please remove if not used

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+    """BFN Montara Platform-specific Component class"""
+
+    #bf_model = get_model()
+    bpcp = BFPlatformComponentsParser(is_modular_chassis=False,  bf_model="Wedge100BF-32X-O-AC-F-BF")
```
it should be generic plugin for Montara/Maveriks/Newport
please use sonic_py_common library to get platform type
e.g., https://github.com/Azure/sonic-buildimage/blob/3bbe562bcaeb3e16fec86f27c185d30b5b44cf6e/platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/fan_drawer.py#L87

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,394 @@
+        except IndexError as e:
+            raise IndexError(str(e) + " - Error: No components found in plaform_components.json")
+        
+        if(self.name=="BMC"):
```
one space after `if` and `elif`

Commit:
[BFN] Implementation API for platform component
            SONiC has a concept of "platform components"
            this may include - CPLD, FPGA, BIOS, BMC, etc.

            These changes are needed to read the version of the BIOS and BMC component.

            What I did
                    Create components.py module
                    Add funcion for reading componet version to thrift interface
            How I did it
                    The previous implementaion didn't have platform components API, so fwutil return an empty list.
                    After implementation of the platform component API, we have actual list of platform components and firmware versions

            How to verify it
                    Run manually 'fwutil show status' or run unit tests

            Previous command output
                    Chassis                   Module    Component    Version    Description
                    ------------------------  --------  -----------  ---------  -------------

            New command output
                    Chassis                   Module    Component    Version    Description
                    ------------------------  --------  -----------  ---------  -------------
                    Chassis1                  N/A       BIOS         1.2.3     Chassis BIOS
                                                        BMC          5.1       Chassis BMC

Signed-off-by: Taras Keryk <tarasx.keryk@intel.com>

Commit:
[BFN] Implementation API for platform component
    get chassis name from json

Commit:
[BFN] Implementation API for platform component
      Updated platform and platrom_components json

akokhan commented on platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py:
```diff
@@ -0,0 +1,338 @@
+    Returns:
+    A string containing the firmware version of the BMC
+    """
+    ver="N/A"
```
please add spaces `ver = "N/A"`

Commit:
[BFN] Implementation API for platform component
      Fixed spaces in component.py
